tests/framework/static_analysis.py

```diff
@@ -0,0 +1,618 @@
+                            "Could not resolve any argument for syscall wrapper %s in function %s",
+                            called,
+                            fn.name,
+                        )
+
+                    # If the wrapper performs some transformation of an argument, apply it.
+                    # It'd be cool to determine these automatically via back propagation or something,
+                    # but that's a fairly complex task, and we only have a single syscall wrapper that needs this
+                    for arg, modifier in (
+                        SPECIFIC_SYSCALL_WRAPPERS[called]
+                        .get("arg_transform", {})
+                        .items()
+                    ):
+                        syscall_args[arg] = modifier(syscall_args[arg])
+                else:
+                    continue
+            else:
+                continue
+
+            # This gets triggered in the __lockfile function on x86_64 (syscall number is loader before a branching instruction,
+            # but if the branch is not taken, linear execution will eventually hit a ret. So during backpropagation we
+            # would need to skip the section of assembly between "jmp" and "ret", but our script doesn't do anything
+            # sophisticated like that and thus instead tries to analyse this branch where the syscall number register
+            # gets clobbered, and it eventually hits a "pop" which it doesnt understand). The syscall in question is
+            # "futex", and we call that one a million time elsewhere anyway.
+            #
+            # See: https://github.com/kraj/musl/blob/ffb23aef7b5339b8c3234f4c6a93c488dc873919/src/stdio/__lockfile.c#L4
+            if isinstance(syscall_nr, Exception):
+                logger.warning(
+                    "Failed to resolve syscall number for instruction %s in function %s: %s",
+                    insn,
+                    fn.name,
+                    syscall_nr,
+                )
+                continue
+
+            syscall_name = seccomp.resolve_syscall(
+                arch.seccomp_arch, syscall_nr
+            ).decode("utf-8")
+            if syscall_name not in found_syscalls:
+                found_syscalls[syscall_name] = []
+
+            found_syscalls[syscall_name].append(
+                [None if isinstance(arg, Exception) else arg for arg in syscall_args]
+            )
+
+    return found_syscalls
+
+
+def load_seccomp_rules(seccomp_path: Path):
+    """Loads seccomp rules from the given file, and presents them as a dictionary
+    mapping syscalls to a list of individual filters. Each individual filter
+    describes some restriction of the arguments that are allowed to be passed
+    to the syscall."""
+    filters = json.loads(seccomp_path.read_text("utf-8"))
+
+    all_filters = (
+        filters["vcpu"]["filter"] + filters["vmm"]["filter"] + filters["api"]["filter"]
+    )
+    allowlist = {}
```

Comment:
super nit: you can use [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict)

Reply:
Ahhh, I keep forgetting this exists. Changed :)